clang/lib/Sema/SemaDecl.cpp

```diff
@@ -6347,6 +6347,26 @@ NamedDecl *Sema::ActOnVariableDeclarator(
     return nullptr;
   }
 
+  if (R->isSamplerT()) {
+    // OpenCL v1.2 s6.9.b p4:
```

Comment:
This comment looks stange. We implement some **OpenCL** restrictions but not only for OpenCL.

Reply:
Actually I can't find any restrictions for sampler in SYCL spec. I think we can leave comment describes that SYCL re-uses OpenCL types so these restrictions applied for SYCL too but it's not necessary.

Reply:
NOTE: the restriction is enabled for **all** languages, which allows sampler type. I don't think explicitly mentioning SYCL here is good idea.

Reply:
Okay. So, why we are explicitly mentioning OpenCL here?

Reply:
This is the place where the behavior of this type is defined.